backend/src/sync.js

```diff
@@ -348,10 +354,19 @@ async function syncOldNearcoreState() {
   let oldestSyncedBlockHeight = 0;
   if (oldestSyncedBlock !== null) {
     oldestSyncedBlockHeight = oldestSyncedBlock.height;
+    console.debug("-------------------------------------------------------");
     console.debug(`The oldest synced block is #${oldestSyncedBlockHeight}`);
   }
-
-  await syncNearcoreBlocks(oldestSyncedBlockHeight - 1, 1);
+  let syncOldBlockHeight = oldestSyncedBlockHeight;
+  const batch = 10000;
+  do {
+    console.debug(
+      `-------The oldest synced block is #${syncOldBlockHeight}-------`
+    );
+    let lowBlockHeight = Math.max(syncOldBlockHeight - batch, 0);
+    await syncNearcoreBlocks(syncOldBlockHeight - 1, lowBlockHeight);
+    syncOldBlockHeight -= batch;
+  } while (syncOldBlockHeight > 0);
```

Comment:
1. `syncNearcoreBlocks` already does that in batches, so there is no need to implement this batching here.
2. It seems that the fix is just replace:

```diff
-  await syncNearcoreBlocks(oldestSyncedBlockHeight - 1, 1);
+  await syncNearcoreBlocks(oldestSyncedBlockHeight - 1, 0);
```